camelot/view/art.py

```diff
@@ -118,6 +120,104 @@ def __init__(self, image):
     def getQIcon(self):
         return QtGui.QIcon(QtGui.QPixmap.fromImage(self.image))
 
+
+class FontIconEngine(QtGui.QIconEngine):
+
+    def __init__(self):
+        super().__init__()
+        self.font_family = 'FontAwesome'
+        self.code = 'X'
+        self.color = QtGui.QColor()
+
+    def paint(self, painter, rect, mode, state):
+        """
+        :param painter: a :class:`QtGui.QPainter` object
+        :param rect: a :class:`QtCore.QRect` object
+        :param mode: a :class:`QtGui.QIcon.Mode` object
+        :param state: a :class:`QtGui.QIcon.State` object
+        """
+        font = QtGui.QFont(self.font_family)
+        drawSize = QtCore.qRound(rect.height() * 0.8)
+        font.setPixelSize(drawSize)
+
+        penColor = QtGui.QColor()
+        if not self.color.isValid():
+            penColor = QtWidgets.QApplication.palette("QWidget").color(QtGui.QPalette.Normal, QtGui.QPalette.ButtonText)
+        else:
+            penColor = self.color
+
+        if mode == QtGui.QIcon.Disabled:
+            penColor = QtWidgets.QApplication.palette("QWidget").color(QtGui.QPalette.Disabled, QtGui.QPalette.ButtonText)
+
+        if mode == QtGui.QIcon.Selected:
+            penColor = QtWidgets.QApplication.palette("QWidget").color(QtGui.QPalette.Active, QtGui.QPalette.ButtonText)
+
+        painter.save()
+        painter.setPen(QtGui.QPen(penColor))
+        painter.setFont(font)
+        painter.drawText(rect, QtCore.Qt.AlignCenter | QtCore.Qt.AlignVCenter, self.code)
+        painter.restore()
+
+    def pixmap(self, size, mode, state):
+        """
+        :param size: a :class:`QtCore.QSize` object
+        :param mode: a :class:`QtGui.QIcon.Mode` object
+        :param state: a :class:`QtGui.QIcon.State` object
+        """
+        pix = QtGui.QPixmap(size)
+        pix.fill(QtCore.Qt.transparent)
+
+        painter = QtGui.QPainter(pix)
+        self.paint(painter, QtCore.QRect(QtCore.QPoint(0, 0), size), mode, state)
+        painter.end()
+
+        return pix
+
+
+class FontIcon:
+
+    _name_to_code = None
+
+    def __init__(self, name, pixmap_size=32):
+        """
+        The pixmap size is only used when calling getQPixmap().
+        """
+        self._name = name
+        self._pixmap_size = pixmap_size
+
+        if FontIcon._name_to_code is None:
+            FontIcon._load_name_to_code()
+
+    @staticmethod
+    def _load_name_to_code():
+        path = file_('awesome/name_to_code.json')
```

Comment:
hier moet je mee oppassen : resource_filename, gaat een file uit de 'egg' extraheren, op disk zetten, en het pad teruggeven naar die geextraheerde file.

resource_string geeft de geunzipte content vd file terug

dat laatste is dus een puur in memory operatie, terwijl dat eerste mogelijk pokke traag is
(lees temp folders in windows die op een gesharde disk staan)

Reply:
Aangepast, zie nieuwe push.